lbry/lbry/extras/daemon/Components.py

```diff
@@ -136,86 +146,60 @@ def component(self):
-        await first_connection
-        remote_height = await net.get_server_height()
-        await net.stop()
-        return remote_height
+    async def get_download_height(self):
+        async with utils.aiohttp_request('HEAD', HEADERS_URL) as response:
+            if response.status != 200:
+                log.warning("Header download error: %s", response.status)
+                return 0
+            return response.content_length // HEADER_SIZE
 
     async def should_download_headers_from_s3(self):
         if self.conf.blockchain_name != "lbrycrd_main":
             return False
-        self._check_header_file_integrity()
         s3_headers_depth = self.conf.s3_headers_depth
         if not s3_headers_depth:
             return False
-        local_height = self.local_header_file_height()
-        remote_height = await self.get_remote_height()
+
+        local_height = self.headers.height
+        try:
+            remote_height = await self.get_download_height()
+        except OSError:
+            log.warning("Failed to download headers using https.")
+            return False
         log.info("remote height: %i, local height: %i", remote_height, local_height)
         if remote_height > (local_height + s3_headers_depth):
             return True
         return False
 
-    def _check_header_file_integrity(self):
-        # TODO: temporary workaround for usability. move to txlbryum and check headers instead of file integrity
-        if self.conf.blockchain_name != "lbrycrd_main":
-            return
-        hashsum = sha256()
-        checksum_height, checksum = HEADERS_FILE_SHA256_CHECKSUM
-        checksum_length_in_bytes = checksum_height * HEADER_SIZE
-        if self.local_header_file_size() < checksum_length_in_bytes:
-            return
-        with open(self.headers_file, "rb") as headers_file:
-            hashsum.update(headers_file.read(checksum_length_in_bytes))
-        current_checksum = hashsum.hexdigest()
-        if current_checksum != checksum:
-            msg = f"Expected checksum {checksum}, got {current_checksum}"
-            log.warning("Wallet file corrupted, checksum mismatch. " + msg)
-            log.warning("Deleting header file so it can be downloaded again.")
-            os.unlink(self.headers_file)
-        elif (self.local_header_file_size() % HEADER_SIZE) != 0:
-            log.warning("Header file is good up to checkpoint height, but incomplete. Truncating to checkpoint.")
-            with open(self.headers_file, "rb+") as headers_file:
-                headers_file.truncate(checksum_length_in_bytes)
-
     async def start(self):
         if not os.path.exists(self.headers_dir):
             os.mkdir(self.headers_dir)
         if os.path.exists(self.old_file):
             log.warning("Moving old headers from %s to %s.", self.old_file, self.headers_file)
             os.rename(self.old_file, self.headers_file)
+        await self.headers.open()
+        self.headers.repair()
```

Comment:
`repair()` should probably be `async` as well, just from API symmetry point of view (and in case later it actually becomes async). it's safe in Python to mark a method as `async` even if there are no `await`s in it